addons/website_sale/views/templates.xml

```diff
@@ -2291,7 +2291,7 @@
             >
                 <!-- We don't allow public users to have multiple delivery addresses. -->
                 <t t-if="is_invoice">
-                    <t t-set="new_address_href" t-valuef="/shop/address?address_type=billing"/>
+                    <t t-set="new_address_href" t-valuef="/shop/address?address_type=billing&amp;use_delivery_as_billing={{use_delivery_as_billing}}"/>
```

Comment:
After testing and giving it more thought, it doesn't seem right to always provide this value here.
If you have deliverable products, the billing addresses won't be shown unless you uncheck the `use_delivery_as_billing`, but the href was already set, while considering `use_delivery_as_billing` to be true.
Also, you get a warning which is not even true when creating a new billing address.
![image](https://github.com/user-attachments/assets/6fe69486-39f7-49a4-8f48-d54b0c94317d)

Is it even necessary to change that here? :thinking: 
Considering the diff you have in the enterprise branch, you're only interested in `use_delivery_as_billing` when it's a delivery address, not when it's a billing one.

Reply:
@Feyensv I'll remove it since it doesn't seem to change anything anyway but there is some strange behavior around `use_delivery_as_billing`, its value is always True when entering the address form page no matter what.  I wanted to make sure I didn't add the fields everywhere with no regard for the type of address but I guess that's what we're doing then ?

So the fields will show up on the first address (Whether it's billing or delivery) and regardless of `use_delivery_as_billing` which according to your previous message seems fine I guess.

Reply:
I don't mind if it's always True for billing addresses, as it should only be considered for delivery addresses, but it shouldn't be always True for delivery ones.  It's probably because the href is not updated when the value is modified.

Reply:
Got it, let me know when you have news

Reply:
Tested right now, and the href is correctly updated for delivery addresses when `use_delivery_as_billing` is unchecked, I don't think there is an issue here.

Reply:
Alright then, I'll take your word for it. Thanks !